internal/setup/setup.go

```diff
@@ -375,27 +395,48 @@ func (yp *yamlPath) SameContent(other *yamlPath) bool {
 }
 
 type yamlArch struct {
-	list []string
+	List []string
 }
 
 func (ya *yamlArch) UnmarshalYAML(value *yaml.Node) error {
 	var s string
 	var l []string
 	if value.Decode(&s) == nil {
-		ya.list = []string{s}
+		ya.List = []string{s}
 	} else if value.Decode(&l) == nil {
-		ya.list = l
+		ya.List = l
 	} else {
 		return fmt.Errorf("cannot decode arch")
 	}
 	// Validate arch correctness later for a better error message.
 	return nil
 }
 
+func (ya yamlArch) MarshalYAML() (interface{}, error) {
+	if len(ya.List) == 1 {
+		return ya.List[0], nil
+	}
+	return ya.List, nil
+}
+
+type yamlMode uint
+
```

Comment:
```go
var _ yaml.Marshaler = yamlArch{}
....
var _ yaml.Marshaler = yamlMode{}
```
Please add the assertions for all of them in the respective places.

Reply:
Done.